stan/math/opencl/kernel_generator/select.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef STAN_MATH_OPENCL_KERNEL_GENERATOR_SELECT_HPP
+#define STAN_MATH_OPENCL_KERNEL_GENERATOR_SELECT_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/opencl/matrix_cl_view.hpp>
+#include <stan/math/opencl/kernel_generator/type_str.hpp>
+#include <stan/math/opencl/kernel_generator/name_generator.hpp>
+#include <stan/math/opencl/kernel_generator/operation_cl.hpp>
+#include <stan/math/opencl/kernel_generator/as_operation_cl.hpp>
+#include <stan/math/opencl/kernel_generator/is_valid_expression.hpp>
+#include <stan/math/opencl/kernel_generator/common_return_scalar.hpp>
+#include <set>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+namespace stan {
+namespace math {
+
+/**
+ * Represents a selection operation in kernel generator expressions.
+ * @tparam Derived derived type
+ * @tparam T_condition type of condition
+ * @tparam T_then type of then expression
+ * @tparam T_else type of else expression
+ */
+template <typename T_condition, typename T_then, typename T_else>
+class select_ : public operation_cl<select_<T_condition, T_then, T_else>,
+                                    common_scalar_t<T_then, T_else>,
+                                    T_condition, T_then, T_else> {
+ public:
+  using Scalar = common_scalar_t<T_then, T_else>;
+  using base = operation_cl<select_<T_condition, T_then, T_else>, Scalar,
+                            T_condition, T_then, T_else>;
+  using base::var_name;
+
+ protected:
+  using base::arguments_;
+
+ public:
+  /**
+   * Constructor
+   * @param condition condition expression
+   * @param then then expression
+   * @param els else expression
+   */
+  select_(T_condition&& condition, T_then&& then, T_else&& els)  // NOLINT
+      : base(std::forward<T_condition>(condition), std::forward<T_then>(then),
+             std::forward<T_else>(els)) {
+    if (condition.rows() != base::dynamic && then.rows() != base::dynamic) {
+      check_size_match("select", "Rows of ", "condition", condition.rows(),
+                       "rows of ", "then", then.rows());
+    }
+    if (condition.cols() != base::dynamic && then.cols() != base::dynamic) {
+      check_size_match("select", "Columns of ", "condition", condition.cols(),
+                       "columns of ", "then", then.cols());
+    }
+
+    if (condition.rows() != base::dynamic && els.rows() != base::dynamic) {
+      check_size_match("select", "Rows of ", "condition", condition.rows(),
+                       "rows of ", "else", els.rows());
+    }
+    if (condition.cols() != base::dynamic && els.cols() != base::dynamic) {
+      check_size_match("select", "Columns of ", "condition", condition.cols(),
+                       "columns of ", "else", els.cols());
+    }
+  }
+
+  /**
+   * generates kernel code for this expression.
```

Comment:
```suggestion
   * generates kernel code for a ternary operation
```

Reply:
I would prefere to leave this one as it is.

Reply:
What about just going with "for the select operation"?

Reply:
done